k2/csrc/rm_epsilon_test.cu

```diff
@@ -318,4 +320,59 @@ TEST(RmEpsilon, ComputeEpsilonClosureSimple) {
   // TODO(haowen): add random tests
 }
 
+TEST(RmEpsilon, RemoveEpsilonsIterativeTropicalSimple) {
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    std::string s1 = R"(0 1 1 1
+    1 2 0 1 
+    1 3 2 1 
+    2 3 3 1 
+    3 4 4 1 
+    3 5 5 1 
+    4 5 6 1 
+    4 6 7 1
+    5 6 0 1
+    5 7 -1 0
+    6 7 -1 0
+    7
+  )";
+    std::string s2 = R"(0 1 0 1
+    1 2 0 1 
+    2 3 0 1 
+    3 4 4 1 
+    3 5 -1 1 
+    4 5 -1 1
+    5
+  )";
+    Fsa fsa1 = FsaFromString(s1);
+    Fsa fsa2 = FsaFromString(s2);
+    Fsa *fsa_array[] = {&fsa1, &fsa2};
+    FsaVec fsa_vec = CreateFsaVec(2, &fsa_array[0]);
+    fsa_vec = fsa_vec.To(context);
+
+    FsaVec dest;
+    Ragged<int32_t> arc_map;
+    RemoveEpsilonsIterativeTropical(fsa_vec, &dest, &arc_map);
+    EXPECT_EQ(dest.NumAxes(), 3);
+    EXPECT_EQ(arc_map.NumAxes(), 2);
+    K2_LOG(INFO) << dest;
+    K2_LOG(INFO) << arc_map;
+    Array1<int32_t> properties;
+    int32_t p;
+    GetFsaVecBasicProperties(dest, &properties, &p);
+    EXPECT_EQ(p & kFsaPropertiesEpsilonFree, kFsaPropertiesEpsilonFree);
+    bool log_semiring = false;
+    float beam = std::numeric_limits<float>::infinity();
+    fsa_vec = fsa_vec.To(GetCpuContext());
+    dest = dest.To(GetCpuContext());
+    // as both fsa_vec and dest would be top-sorted, we can call
+    // IsRandEquivalent to check the result, but for now top-sorted Fsas, how
+    // can we check the result?
```

Comment:
And here, any suggestion? our current implementation `IsRandEquivalent` requires the inputs are top-sorted as it will compute forward/backward weights in it. For non-top-sorted tests, I think we can call ` IsRandEquivalentUnweighted` to check the sequence, but have no idea now about checking the weights.

Reply:
Mm, theres no fundamental reason why IsRandEquivalent should not be able to operate on non-top-sorted input I believe (?).  Would have to check.   Perhaps we chould do tests on top-sorted input to RmEpsilon, it will stay top-sorted.

Reply:
as I said above, it's becasue currently we call `shortestDistance` in `IsRandEquivalent` to get the score of the sequence, which calls `computeForwardWeights` internally and then it requires the input top-sorted.

Reply:
I think I just merged a change so IsRandEquivalent will work on non-top-sortable input as long as log_semiring==false.

Reply:
sorry, I just tried to find but seems there's no such one，can you paste the link? BTW, so if there's a cyclic, how do you compute the forward weights (even for tropical), using relax variable stuff?

Reply:
It's a change to the existing algorithm.
It uses a queue; you re-add something to the queue if the score changes.